### PR TITLE
feat: add jinja datestamp output

### PR DESCRIPTION
**Pull request by deenine**
Signed-off-by: Doug Chivers <doug.chivers@uk.ibm.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [ ] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

## Summary

Add a md_datestamp Jinja tag to add a date stamp to the markdown output. 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by deenine** [comment]
This requires tests, otherwise ready for review.

File: docs/trestle_author_jinja.md [hunk 1]
```diff
@@ -111,8 +111,16 @@ Trestle provides to custom jinja tags for use specifically with markdown: `mdsec
...
+
    1. `{% mdsection_include 'test_markdown.md' '# Header we want' %}`
    1. The heading level argument adjusts to (based on the number of hashes) the most significant heading in the chosen section, if headings exist.
+
+`md_datestamp` inserts a date stamp into the output. By default the date is in the format '%m/%d/%Y', e.g. '12/28/2001' and is followed by a double newline to prevent subsequent headings failing to parse correctly. E.g: `{% md_datestamp %}` results in a date in the format '12/28/2021' being inserted. `md_datestamp` can be used with the following optional keyword arguments:
+
+1. `format` where a python [datetime strftime format string](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes) is provided to format the output. E.g. `{% md_datestamp format='%B %d, %Y' %}` results in `December 28, 2021` being inserted.
+1. `newline` is a boolean to control the addition of a double newline after the inserted date string. For example `{% md_datestamp newline=false %}` inserts a date in the default format, without additional newlines.
```
Comment (fsuits) [inline docs/trestle_author_jinja.md:126]: Isn't %Y-%m-%d a more common default?  I realize the user can change it.  Trestle currently mainly uses that if you look for strptime and strftime.

File: trestle/core/jinja.py [hunk 2]
```diff
@@ -177,3 +178,53 @@ def parse(self, parser):
...
+                if parser.stream.look().type == lexer.TOKEN_ASSIGN or parser.stream.look().type == lexer.TOKEN_STRING:
+                    kwargs = {}
+                continue
+
+        if kwargs is not None:
+            if 'format' in kwargs and type(kwargs['format'] is str):
+                date_string = date.today().strftime(kwargs['format'])
+            else:
+                date_string = date.today().strftime('%m/%d/%Y')
+            if 'newline' in kwargs and kwargs['newline'] is False:
```
Comment (fsuits) [inline trestle/core/jinja.py:220]: The repeated references to the timestring might be better as a constant - const.JINGA_DATESTAMP_FORMAT or something.  There are many refs to it here and the test.

**Review by fsuits — APPROVED** [review]
LGTM with my suggestion to make a constant for the date format and consider other defaults.  I don't know the details of the jinga code itself so I defer to Ekaterina for her final review.

File: trestle/core/jinja.py (hunk 2 above)
Comment (deenine) [reply trestle/core/jinja.py]: done

**Review by enikonovad — APPROVED** [review]
LGTM

**Pull request merged**